app/controllers/products_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ProductsController < ApplicationController
+
+	def index
+    @vendor = Vendor.find(params[:vendor_id])
+    @products = @vendor.products
+  end
+
+  def show
+    @products = Product.find(params[:vendor_id])
+  end
+
+  def new
+    @vendor = Vendor.find(params[:vendor_id])
+    @product = @vendor.products.new
+  end
+
+  def create
+    Product.create(product_create_params[:product])
+    redirect_to vendor_products_path(params[:vendor_id])
+  end
+
+  def edit
+    @product = Product.find(params[:id])
+    @vendor = Vendor.find(params[:vendor_id])
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    @product.update_attributes(product_update_params[:product])
+    redirect_to vendor_products_path
+  end
+
+  def destroy
+  	@product = Product.find(params[:id])
+    @product.destroy
+    redirect_to vendor_products_path
+  end
```

Comment:
You don't need @product to be an instance variable here, since you're not passing it to a view. It can be ```product = Product.find(params[:id])```